config/formats.ts

```diff
@@ -707,6 +707,39 @@ export const Formats: FormatList = [
 			return null;
 		},
 		onValidateTeam(team, f, teamHas) {
+			// 2 Ability Clause
+			const abilityTable = new Map<string, number>();
+			const base: {[k: string]: string} = {
+				airlock: 'cloudnine',
+				battlearmor: 'shellarmor',
+				clearbody: 'whitesmoke',
+				dazzling: 'queenlymajesty',
+				emergencyexit: 'wimpout',
+				filter: 'solidrock',
+				gooey: 'tanglinghair',
+				insomnia: 'vitalspirit',
+				ironbarbs: 'roughskin',
+				libero: 'protean',
+				minus: 'plus',
+				moxie: 'chillingneigh',
+				powerofalchemy: 'receiver',
+				propellertail: 'stalwart',
+				teravolt: 'moldbreaker',
+				turboblaze: 'moldbreaker',
+			};
+			for (const set of team) {
+				let ability = this.toID(set.ability.split('0')[0]);
+				if (!ability) continue;
+				if (ability in base) ability = base[ability] as ID;
+				if ((abilityTable.get(ability) || 0) >= 2) {
+					return [
+						`You are limited to two of each ability by 2 Ability Clause.`,
+						`(You have more than two ${this.dex.abilities.get(ability).name} variants)`,
+					];
+				}
+				abilityTable.set(ability, (abilityTable.get(ability) || 0) + 1);
+			}
```

Comment:
i would encompass this in an `if` statement to check that this is only applied if the ruleTable has `2abilityclause`